crates/rolldown_common/src/chunk/mod.rs

```diff
@@ -172,12 +172,7 @@ impl Chunk {
 
     let filename = filename_template.render(Some(chunk_name), None, hash_replacer).into();
 
-    let name = if has_hash_pattern {
-      make_unique_name(&filename, used_name_counts);
-      filename
```

Comment:
Here the using `make_unique_name` to deconflict filename, but not using it retured value as filename, it is an unexpected error cc @hyf0

Reply:
I remember I did it on purpose, but I can't think of why.

> but not using it retured value

Because rollup didn't use it. It makes the expected `name-xxxxx.js` becomes `name2-xxxx.js`. With hash, the final filename should not be conflicted.

Reply:
let's always deconflict to aovid conflict if enable hash. I think it maybe happend.